debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnection.java

```diff
@@ -579,16 +580,25 @@ public String buildReselectColumnQuery(TableId tableId, List<String> columns, Li
             return super.buildReselectColumnQuery(oracleTableId, columns, keyColumns, source);
         }
 
-        return String.format("SELECT %s FROM (SELECT * FROM %s AS OF SCN %s) WHERE %s",
+        return String.format("SELECT %s FROM (SELECT * FROM %s AS OF SCN ?) WHERE %s",
                 columns.stream().map(this::quotedColumnIdString).collect(Collectors.joining(",")),
                 quotedTableIdString(oracleTableId),
-                commitScn,
                 keyColumns.stream().map(key -> key + "=?").collect(Collectors.joining(" AND ")));
     }
 
     @Override
-    public Map<String, Object> reselectColumns(String query, TableId tableId, List<String> columns, List<Object> bindValues) throws SQLException {
-        return optionallyDoInContainer(() -> super.reselectColumns(query, tableId, columns, bindValues));
+    public Map<String, Object> reselectColumns(String query, TableId tableId, List<String> columns, List<Object> bindValues, Struct source) throws SQLException {
+        final String commitScn = source.getString(SourceInfo.COMMIT_SCN_KEY);
+        final List<Object> values;
+        if (Strings.isNullOrEmpty(commitScn)) {
+            values = bindValues;
+        }
+        else {
+            values = new ArrayList<>(bindValues.size() + 1);
+            values.add(commitScn);
+            values.addAll(bindValues);
+        }
```

Comment:
Hi @jchipmunk, good call.  I suppose the only thing I particularly don't like is the notion that we use COMMIT_SCN_KEY in both methods now to so very similar things, but I'm not sure we can clean this up anymore without there being a larger refactor involved, so likely best to leave as-is for now.

Reply:
Hi @Naros, I agree, but did not dare to change the current methods much.